tests/unit/pluginhandler/test_scriptlets.py

```diff
@@ -29,77 +29,100 @@
+        self.assertThat(getattr(metadata, self.getter)(), Equals('test-value'))
 
-    def test_set_version_in_build(self):
+    def test_set_in_build(self):
         handler = self.load_part('test_part', part_properties={
-            'override-build': 'snapcraftctl set-version override-version'
+            'override-build': 'snapcraftctl {} test-value'.format(
+                self.setter)
         })
 
         handler.pull()
         handler.build()
         metadata = handler.get_build_state().scriptlet_metadata
-        self.assertThat(metadata.get_version(), Equals('override-version'))
+        self.assertThat(getattr(metadata, self.getter)(), Equals('test-value'))
         self.assertFalse(handler.get_pull_state().scriptlet_metadata)
 
-    def test_set_version_in_stage(self):
+    def test_set_in_stage(self):
         handler = self.load_part('test_part', part_properties={
-            'override-stage': 'snapcraftctl set-version override-version'
+            'override-stage': 'snapcraftctl {} test-value'.format(
+                self.setter)
         })
 
         handler.pull()
         handler.build()
         handler.stage()
         metadata = handler.get_stage_state().scriptlet_metadata
-        self.assertThat(metadata.get_version(), Equals('override-version'))
+        self.assertThat(getattr(metadata, self.getter)(), Equals('test-value'))
         self.assertFalse(handler.get_pull_state().scriptlet_metadata)
         self.assertFalse(handler.get_build_state().scriptlet_metadata)
 
-    def test_set_version_in_prime(self):
+    def test_set_in_prime(self):
         handler = self.load_part('test_part', part_properties={
-            'override-prime': 'snapcraftctl set-version override-version'
+            'override-prime': 'snapcraftctl {} test-value'.format(
+                self.setter)
         })
 
         handler.pull()
         handler.build()
         handler.stage()
         handler.prime()
         metadata = handler.get_prime_state().scriptlet_metadata
-        self.assertThat(metadata.get_version(), Equals('override-version'))
+        self.assertThat(getattr(metadata, self.getter)(), Equals('test-value'))
         self.assertFalse(handler.get_pull_state().scriptlet_metadata)
         self.assertFalse(handler.get_build_state().scriptlet_metadata)
         self.assertFalse(handler.get_stage_state().scriptlet_metadata)
 
 
-class ScriptletSetVersionErrorTestCase(unit.TestCase):
+class ScriptletMultipleSettersErrorTestCase(unit.TestCase):
 
     scriptlet_scenarios = [
-        ('override-pull', {'override_pull': 'snapcraftctl set-version 1'}),
-        ('override-build', {'override_build': 'snapcraftctl set-version 2'}),
-        ('override-stage', {'override_stage': 'snapcraftctl set-version 3'}),
-        ('override-prime', {'override_prime': 'snapcraftctl set-version 4'}),
+        ('override-pull', {'override_pull': 'snapcraftctl {setter} 1'}),
```

Comment:
taking it to the next level ey?

Reply:
Yeah man. Show me another use of scenarios that is this awesome.